addOns/grpc/src/main/java/org/zaproxy/addon/grpc/internal/HttpPanelGrpcView.java

```diff
@@ -126,7 +130,11 @@ public void save() {
         try {
             protoBufMessageEncoder.encode(EncoderUtils.parseIntoList(text));
             byte[] encodedMessage = protoBufMessageEncoder.getOutputEncodedMessage();
-            this.model.setData(Base64.getEncoder().encode(encodedMessage));
+            if (decodingMethod == DecoderUtils.DecodingMethod.BASE64_ENCODED) {
+                this.model.setData(Base64.getEncoder().encode(encodedMessage));
+            } else {
+                this.model.setData(encodedMessage);
+            }
```

Comment:
I thought this (the encoding check) was going to be done inside the encoder/decoder. Likely that we will have more places that will need this.

Reply:
but i think the encoder and decoder should be meant to decode or encode protobuff message only, not base64, putting this check inside encoder make it confusing. just a thought 🤔

Reply:
In that case `DecodingMethod` should not live in that class, IMO since that's about/related to decoding it does not hurt to live in that class as well (otherwise we will create a class with just this if?). We should also note that WebSockets is using internal classes, ideally we should provide a consistent API for other add-ons to consume (as it's still alpha it should be fine).

Reply:
Okay I can make a seperate file just for decodingMethod enum what should I name it DecodingMethodUtils😂

Reply:
DecodingMethod is good in DecoderUtils class